### ISSUE TITLE
Allow a message to be sent to update contract configurations

### ISSUE THREAD

**Issue by leeduan** [provenance-io/ats-smart-contract#80]
## Summary

Allow a broadcast to update the contract configurations with a msg instead of requiring a migration

## Problem Definition

Requiring a migrate of the SC to update configurations limits frequency of updating config. This will allow us to change bid/ask attributes or fee percentage, etc. without a wasm install and migrate.

## Proposal

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by ktalley-figure** [comment]
since most of the fields in ContractInfo have an impact on how orders are created/processed, we'll need to determine how existing orders are to be handled for each field. 

additionally, certain fields cannot be changed, i believe:
```
pub struct ContractInfoV3 {
    pub name: String,   <--- cannot change?
    pub bind_name: String,   <--- cannot change
    pub base_denom: String,   <--- cannot change
    pub convertible_base_denoms: Vec<String>,   <--- cannot change
    pub supported_quote_denoms: Vec<String>,   <--- cannot change
    pub approvers: Vec<Addr>,   <--- needs more info
    pub executors: Vec<Addr>,
    pub ask_fee_info: Option<FeeInfo>,
    pub bid_fee_info: Option<FeeInfo>,   <--- needs more info
    pub ask_required_attributes: Vec<String>,   <--- needs more info
    pub bid_required_attributes: Vec<String>,   <--- needs more info
    pub price_precision: Uint128,   <--- cannot change
    pub size_increment: Uint128,   <--- cannot change
}
```
more info needed:

- which address(s) are allowed to make ContractInfo updates?
- `approvers` - if an approver, who previously approved an order, is removed and the order is later canceled/rejected, where should the approved funds go? 
  - currently, approved funds will always be sent back to the address that provided them

- `bid_fee_info` - if fees change, should the existing orders use the rate with which the order was originally submitted? including if fees are removed? if the fee info is removed, existing orders with fees won't have an address to send to. 
  - currently, the order pays fees at the rate with which it was created, even if the rate is later set to 0. also, the contract will reject the execute match if an order has fees, but the fee address does not exist
- `ask_required_attributes` - if an existing order no longer meets the new attribute requirements, how should it be handled?
  - currently, the order is validated once at creation and never again 
- `bid_required_attributes` - same as above

**Comment by leeduan** [comment]
sounds good @ktalley-figure probably a good exercise for a new dev to work on and train up

**Comment by iramiller** [comment]
General Comment: It might be worth considering that contract design really needs to target the smallest and least complex workflow that does the job. Putting in lots of logic that is based on updating security and contract control is a recipe for particularly nasty bugs... as a general rule those flows should be  done at a layer above/off-chain.

**Comment by jtalis** [comment]
addressed in #84
